packages/graphql/src/providers/ThrottleRPCProvider.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Based on: https://github.com/MetaMask/web3-provider-engine/blob/e90af04ee57f5b4f2e33f44630f28d7cd2ec7c72/subproviders/rpc.js
+ * This is a modified subprovider to handle rate limiting.
+ *
+ * Blame: Mike Shultz <shultzm@gmail.com>
+ */
+import fetch from 'cross-fetch'
+import Bottleneck from 'bottleneck/es5'
+import JsonRpcError from 'json-rpc-error'
+import createPayload from 'web3-provider-engine/util/create-payload'
+import SubProvider from 'web3-provider-engine/subproviders/subprovider'
+
+const MAX_RETRIES = 3
+const BUFFER_MS = 5
+// TODO Expand on this maybe?
+const METHOD_PRIORITY = {
+  eth_blockNumber: 9,
+  eth_getBlockByNumber: 7,
+  eth_getLogs: 4,
+  eth_call: 1
+}
+
+/**
+ * Send a JSON-RPC request
+ * @param url {string} The destination for the request
+ * @param payload {object} The JSON-RPC body
+ * @returns {object} The JS repr of the JSON response data
+ */
+async function sendRequest(url, payload) {
+  let response = null
+  let error = null
+  try {
+    response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(payload)
+    })
+  } catch (err) {
+    error = err
+  }
+
+  if (error) {
+    throw new JsonRpcError.InternalError(error)
+  }
+
+  // check for error code
+  switch (response.status) {
+    case 405:
+      throw new JsonRpcError.MethodNotFound()
+    case 504: // Gateway timeout
+      return (function() {
```

Comment:
My JS fu is terribly weak. Why do we need to use a function here ?

Reply:
More artifacts from the original source...  It was originally returning with a call to `end()` with this error, but we want to throw here in this new configuration.  I'll rip this stuff out.  Thanks